crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -3,14 +3,114 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use syn::{parse::ParseStream, Attribute, Ident, Result, Token, Type};
+use std::collections::HashSet;
+
+use syn::{
+    parse::{Error, ParseStream},
+    punctuated::Punctuated,
+    spanned::Spanned,
+    Attribute, Expr, Ident, Meta, MetaNameValue, Path, Result, Token, Type,
+};
+
+use quote::format_ident;
+
+/// An optional identifier of the functions passed with flags
+/// e.g. read = my_getter, IdentFlag would be used to store my_getter
+type IdentFlag = Option<Ident>;
+
+/// Struct for storing the flags provided for a QProperty and their optional identifiers ([IdentFlag])
+#[derive(Debug)]
+pub struct QPropertyFlags {
+    read: IdentFlag, // TODO: maybe change this to better represent the data
+    write: Option<IdentFlag>,
+    notify: Option<IdentFlag>,
+}
+
+impl QPropertyFlags {
+    pub fn new() -> Self {
+        Self {
+            read: None,
+            write: None,
+            notify: None,
+        }
+    }
+
+    pub fn all_flags() -> Self {
+        Self {
+            read: None,
+            write: Some(None),
+            notify: Some(None),
+        }
+    }
+
+    fn set_field_by_string(&mut self, key: String, value: IdentFlag) -> Result<String> {
+        match key.as_str() {
+            "read" => self.read = value,
+            "write" => self.write = Some(value),
+            "notify" => self.notify = Some(value),
+            _ => {
+                return Err(Error::new(
+                    format_ident!("{}", key).span(), // TODO: check if this is an acceptable form of erroring for non Syn functions
+                    "Invalid flag passed, must be one of read, write, notify",
+                ));
+            }
+        };
+        Ok(key) // Return the string back up for checking which flags were set
+    }
+
+    fn add_from_meta(&mut self, meta_value: Meta) -> Result<String> {
+        match meta_value {
+            Meta::Path(path) => {
+                let ident: String = parse_path_to_ident(&path).to_string();
+
+                Ok(self.set_field_by_string(ident, None)?)
+            }
+            Meta::NameValue(name_value) => {
+                let kv_pair = parse_meta_name_value(&name_value)?;
+
+                let ident: String = kv_pair.0.to_string();
+                let value: IdentFlag = Some(kv_pair.1);
+
+                Ok(self.set_field_by_string(ident, value)?)
+            }
+            _ => Err(Error::new(
+                meta_value.span(),
+                "Invalid syntax, flags must be specified as either `read` or `read = my_getter`",
+            )),
+        }
+    }
+}
 
 /// Describes a single Q_PROPERTY for a struct
 pub struct ParsedQProperty {
     /// The [syn::Ident] of the property
     pub ident: Ident,
     /// The [syn::Type] of the property
     pub ty: Type,
+    /// Property flag collection
+    pub flags: QPropertyFlags,
+}
+
+fn parse_path_to_ident(path: &Path) -> Ident {
+    path.segments[0].ident.clone()
+}
```

Comment:
hmm this assumes that there is only one part of the path

Reply:
see longer comment above about using `path_compare_str`

Reply:
I think with the way it's being parsed in, there can only be one path segment, because they have to comma separated for the syn::Punctuated parsing, and if it was a List or NameValue it gets parsed properly?

Reply:
So now that you have the right `Path` coming in, instead use `Path::require_ident() -> Result<Ident>`, you could probably just remove this function and instead change it's usages?

eg instead of
```
Meta::Path(path) => Ok(self.set_field_by_ident(parse_path_to_ident(&path), None)?),
```
it could be
```
Meta::Path(path) => Ok(self.set_field_by_ident(path.require_ident()?, None)?),
```

You just might need to change some places to accept `&Ident` instead of `Ident` where possible, like `set_field_by_ident` can ? And other places that cannot can use `to_owned()` or `clone()`

https://docs.rs/syn/latest/syn/struct.Path.html#method.require_ident

Reply:
done